packages/vite/src/node/plugins/asset.ts

```diff
@@ -77,20 +81,21 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
 
       id = id.replace(urlRE, '$1').replace(/[\?&]$/, '')
       const url = await fileToUrl(id, config, this)
-      return `export default ${JSON.stringify(url)}`
+      // Return a template string so we can use interpolation when replacing __VITE_ASSET__
+      return `export default ${stringifyAsTemplateLiteral(url)}`
```

Comment:
I've been thinking more on this, would it make sense to skip the stringify if the URL starts with `__VITE_ASSET__`? e.g.

```js
`export default ${url.startsWith('__VITE_ASSET__') ? url : JSON.stringify(url)}
```

That way we don't have to use `${}` in the below code to interpolate, and also reduce build output. 

Testing locally, I notice `url` could be `data:image/blablabla` too which we could early out on.

Reply:
Is this a common enough case? I would prefer to avoid these optimizations if possible. Maybe we could also go with the simpler approach at first, and then add these optimizations so there is less noise while we test the main changes?

Reply:
I think that makes sense. We could improve later on as we go. I think it's a common case though for assets that are inlined, but yeah it only brings a tiny perf boost.